File: trunk/java/picocontainer/src/java/org/picocontainer/extras/DecoratingComponentAdapterFactory.java
package org.picocontainer.extras;

import org.picocontainer.ComponentAdapter;
import org.picocontainer.Parameter;
import org.picocontainer.PicoIntrospectionException;
import org.picocontainer.defaults.AssignabilityRegistrationException;
import org.picocontainer.defaults.ComponentAdapterFactory;
import org.picocontainer.defaults.NotConcreteRegistrationException;

public class DecoratingComponentAdapterFactory implements ComponentAdapterFactory {
    private final ComponentAdapterFactory delegate;

    public DecoratingComponentAdapterFactory(ComponentAdapterFactory delegate) {
        this.delegate = delegate;
    }

    public ComponentAdapter createComponentAdapter(Object componentKey,
                                                   Class componentImplementation,
                                                   Parameter[] parameters) throws PicoIntrospectionException, AssignabilityRegistrationException, NotConcreteRegistrationException {
        return delegate.createComponentAdapter(componentKey, componentImplementation, parameters);
    }
}

File: trunk/dotnet/picocontainer/src/Extras/DecoratingComponentAdapterFactory.cs
/*****************************************************************************
 * Copyright (C) PicoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the license.html file.                                                    *
 *                                                                           *
 * Idea by Rachel Davies, Original code by Aslak Hellesoy and Paul Hammant   *
 * C# port by Maarten Grootendorst                                           *
 *****************************************************************************/

using System;

namespace PicoContainer.Extras
{
	
	public class DecoratingComponentAdapterFactory : ComponentAdapterFactory
	{
		private ComponentAdapterFactory theDelegate;
		
		public DecoratingComponentAdapterFactory(ComponentAdapterFactory theDelegate)
		{
			this.theDelegate= theDelegate;
		}
		
		public virtual ComponentAdapter CreateComponentAdapter(object componentKey, System.Type componentImplementation, Parameter[] parameters)
		{
			return theDelegate.CreateComponentAdapter(componentKey, componentImplementation, parameters);
		}
	}
}